src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -2307,6 +2307,10 @@ namespace PurchasedAirManager {
 		if ( PurchAir( PurchAirNum ).PlenumExhaustAirNodeNum > 0 ) {
 			Node( PurchAir( PurchAirNum ).PlenumExhaustAirNodeNum ).MassFlowRate = SupplyMassFlowRate;
 			Node( RecircNodeNum ).MassFlowRate = SupplyMassFlowRate;
+		} else {
+			if ( PurchAir( PurchAirNum ).ZoneExhaustAirNodeNum > 0 && abs( Node( PurchAir( PurchAirNum ).ZoneExhaustAirNodeNum ).MassFlowRate - SupplyMassFlowRate) > rTinyValue ) {
+				Node( PurchAir( PurchAirNum ).ZoneExhaustAirNodeNum ).MassFlowRate = SupplyMassFlowRate;
+			}
```

Comment:
@lgu1234 I am not sure that this solution is general. If there is no exhaust node, the airflow imbalance will still occur. 

An better solution is to *always* set the recirc node mass flow rate equal to the supply flow rate.  The current line 2309
```
Node( RecircNodeNum ).MassFlowRate = SupplyMassFlowRate;
```
should just come out of the above if block and always execute.

Also, [this block](https://github.com/NREL/EnergyPlus/blob/017868e84f0eb0de7711a942c72aebc273bafcc8/src/EnergyPlus/PurchasedAirManager.cc#L2514-L2517) in `CalcPurchAirMixedAir` should be deleted.
a) This function shouldn't be touching any node massflowrates in the first place (or any node conditions at all)
b) The recirc node should always be set to `SupplyMassFlowRate`. Setting it to `RecircMassFlowRate` was just plain wrong.

Reply:
@mjwitte Here is my understanding. There are 3 nodes in the system, zone inlet node, zone exhaust node, and recirc node. 

If we set Node( RecircNodeNum ).MassFlowRate = SupplyMassFlowRate, and the ideal system is a DOAS system, do we have a relief node to handle outdoor air. If we don't have a relief node, here is my vision:
Node( RecircNodeNum ).MassFlowRate = 0.0;
Node( ZoneInletNodeNum ).MassFlowRate = SupplyMassFlowRate;
Node( ExhaustNodeNum ).MassFlowRate = SupplyMassFlowRate;